yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -206,6 +265,126 @@ func (p *ParserIssueDetector) getPLPGSQLIssues(query string) ([]QueryIssue, erro
+// and the function uses topological sorting to ensure all parent metadata is available
+// before copying columns to their children.
+//
+// For each child, it creates a column metadata map if it doesn't exist,
+// and copies any missing columns from each of its parents.
+//
+// This function is used to populate column metadata for both inherited and partitioned tables.
+func (p *ParserIssueDetector) finalizeColumnsFromParentMap(parentMap map[string][]string) {
+	orderedChildren := topoSort(parentMap)
+
+	for _, child := range orderedChildren {
+		parentList, exists := parentMap[child]
+		if !exists || len(parentList) == 0 {
+			continue // This is a root table with no parents
+		}
+		for _, parent := range parentList {
+			parentCols, ok := p.columnMetadata[parent]
+			if !ok {
+				continue // Parent metadata not found
+			}
+			if _, exists := p.columnMetadata[child]; !exists {
+				p.columnMetadata[child] = make(map[string]*ColumnMetadata)
+			}
+			for colName, colMeta := range parentCols {
+				if _, exists := p.columnMetadata[child][colName]; !exists {
+					copy := *colMeta
+					p.columnMetadata[child][colName] = &copy
+				}
+			}
+		}
+	}
+}
+
+// topoSort returns a topological ordering of the keys in a dependency map,
+// ensuring that all parent tables are visited before their children.
+//
+// The input is a map from child table name to a list of parent table names.
+// This supports both partitioned tables (single parent) and inherited tables (multiple parents).
+//
+// It performs a depth-first traversal and returns a slice of table names
+// such that for any entry [child -> parents], all parents appear before the child in the result.
+//
+// Assumes there are no cycles in the dependency map. This should not happen in either partitioned or inherited tables.
+func topoSort(dependencyMap map[string][]string) []string {
+	visited := make(map[string]bool)
+	var result []string
+
+	var dfs func(string)
+	dfs = func(curr string) {
+		if visited[curr] {
+			return
+		}
+		for _, parent := range dependencyMap[curr] {
+			dfs(parent)
+		}
+		visited[curr] = true
+		result = append(result, curr)
+	}
+
+	for child := range dependencyMap {
+		dfs(child)
+	}
+	return result
+}
+
+// finalizeForeignKeyConstraints updates columnMetadata with foreign key details.
+// It iterates through all stored FK constraints and marks the corresponding local columns
+// as foreign keys, populating their referenced table, column, and type information.
+// This is done after all DDL statements have been processed to ensure complete metadata is
+func (p *ParserIssueDetector) finalizeForeignKeyConstraints() {
+	for _, fk := range p.foreignKeyConstraints {
+		for i, localCol := range fk.columnNames {
+			if _, ok := p.columnMetadata[fk.tableName]; !ok {
+				p.columnMetadata[fk.tableName] = make(map[string]*ColumnMetadata)
+			}
+			meta, ok := p.columnMetadata[fk.tableName][localCol]
+			if !ok {
+				meta = &ColumnMetadata{DataType: "unknown"}
```

Comment:
Do we really need `unknown` to be added here? We already have a check to ignore empty type, and we can also skip unknowns as well to avoid any false positives, right?

Reply:
Yeah true. We are can just handle empty type and not raise an issue for them as such.